azurerm/internal/services/network/public_ip_resource.go

```diff
@@ -103,6 +103,18 @@ func resourcePublicIp() *pluginsdk.Resource {
 				}, true),
 			},
 
+			"sku_tier": {
+				Type:             pluginsdk.TypeString,
+				Optional:         true,
+				Default:          string(network.PublicIPAddressSkuTierRegional),
+				ForceNew:         true,
+				DiffSuppressFunc: suppress.CaseDifference,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(network.PublicIPAddressSkuTierGlobal),
+					string(network.PublicIPAddressSkuTierRegional),
+				}, true),
```

Comment:
all new properties should be case-sensitive here, so can we make this:

```suggestion
				ValidateFunc: validation.StringInSlice([]string{
					string(network.PublicIPAddressSkuTierGlobal),
					string(network.PublicIPAddressSkuTierRegional),
				}, false),
```